File: PerfectAng/Perfect/src/app/Classes/Order.ts
import { PersonalProduct } from "./PersonalProduct";
import { ProductToOrder } from "./ProductToOrder";

export class Order{
    constructor(
        public id : number,
        public sumPrice? : number,
        public userId?:number,
        public delivery?:string,
        public date?:Date,
        public pickupDate?:Date,
        public deliveryPrice?:number,
        public delivered?:boolean,
        public personalProducts? : PersonalProduct[],
        public productToOrders? : ProductToOrder[]
        
       ){}
}

File: DAL/Models/Order.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace DAL.Models
{
    public partial class Order
    {
        public Order()
        {
            PersonalProducts = new HashSet<PersonalProduct>();
            ProductToOrders = new HashSet<ProductToOrder>();
        }

        public int Id { get; set; }
        public int UserId { get; set; }
        public string Delivery { get; set; }
        public DateTime Date { get; set; }
        public DateTime PickupDate { get; set; }
        public int DeliveryPrice { get; set; }
        public bool Delivered { get; set; }
        public int SumPrice { get; set; }

        public virtual ICollection<PersonalProduct> PersonalProducts { get; set; }
        public virtual ICollection<ProductToOrder> ProductToOrders { get; set; }
    }
}
